### DISCUSSION TITLE
Integration with spatie/laravel-activitylog: Log when permission  gets assigned to role

### DISCUSSION THREAD

**Discussion by lucacastelnuovo** [spatie/laravel-permission#2303] [Q&A] [answered]
Hi,

My project uses both spatie/laravel-permission and spatie/laravel-activitylog.
By extending the `Spatie\Permission\Models\Role` to `App\Models\Role`, I can log when a role gets created and when it gets deleted.

Does somebody know how I would log when permission gets "given" to a role using `givePermissionTo()`?

---
**App\Models\Role.php**
```php
<?php

namespace App\Models;

use Spatie\Activitylog\LogOptions;
use Spatie\Activitylog\Traits\LogsActivity;
use Spatie\Permission\Models\Role as SpatieRole;

class Role extends SpatieRole
{
    use LogsActivity;

    public function getActivitylogOptions(): LogOptions
    {
        return LogOptions::defaults()
            ->logOnly(['name'])
            ->logOnlyDirty()
            ->dontSubmitEmptyLogs()
            ->setDescriptionForEvent(fn (string $eventName) => "role.$eventName");
    }
}
```

**Comment by erikn69** [comment, marked as answer]
Maybe overwriting `asignRole` method

**Reply by lucacastelnuovo** [reply]
That is a smart option, I will try that and post my solution when I have implemented it.

**Reply by lucacastelnuovo** [reply]
This solution did not work out for me in the end. Because filament relation field did not the trigger the custom methods on my custom role model. Not sure why maybe because filament uses its own relation handling instead of the assignRole/syncRoles methods.

**Reply by haykaf** [reply]
> This solution did not work out for me in the end. Because filament relation field did not the trigger the custom methods on my custom role model. Not sure why maybe because filament uses its own relation handling instead of the assignRole/syncRoles methods.

Did you find a solution for this issue? If so, could you please share it?

Thanks.

**Reply by lucacastelnuovo** [reply]
Hi @aslanutku not yet, but I must say I did not spend much time afterwards trying to solve this. Maybe a lifecycle hook would be best.

**Reply by parallels999** [reply]
Maybe using `withProperties`?
```php
activity()
   ->performedOn($anUserModel)
   ->causedBy($userSession)
   ->withProperties(['givePermissionTo' => $permissions_array])
```

**Reply by lucacastelnuovo** [reply]
> Maybe using `withProperties`?
> 
> ```
> activity()
>    ->performedOn($anUserModel)
>    ->causedBy($userSession)
>    ->withProperties(['givePermissionTo' => $permissions_array])
> ```

How would this work in the filament select field?

**Reply by parallels999** [reply]
I have no idea what it filament  is

**Reply by lucacastelnuovo** [reply]
Filament is https://github.com/filamentphp/filament but I understand that that is out of scope for you @parallels999 .
